offchain/eth-input-reader/src/setup.rs

```diff
@@ -1,88 +1,116 @@
 // (c) Cartesi and individual authors (see AUTHORS)
 // SPDX-License-Identifier: Apache-2.0 (see LICENSE)
 
```

Comment:
I think many functions in this file could be joined into a single one, returning just what the `start` function needs.